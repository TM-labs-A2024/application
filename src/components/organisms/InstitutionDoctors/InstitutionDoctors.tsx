import { HamburgerIcon } from '@chakra-ui/icons'
import { Text, IconButton, Menu, MenuButton, MenuList, MenuItem, Icon } from '@chakra-ui/react'
import RequestsList from '@components/molecules/RequestsList'
import { Doctor as DoctorType, ReactSelectOption, Nurse } from '@src/types'
import { isIOS, isAndroid } from '@utils/index'
import Link from 'next/link'
import React, { useMemo } from 'react'
import { FaUserDoctor } from 'react-icons/fa6'

import Logo from '../../../../public/static/icons/logo.svg'

export default function InstitutionDoctors({
  context: { doctors, specialtiesOptions, nurses }
}: {
  context: { doctors?: DoctorType[]; specialtiesOptions: ReactSelectOption[]; nurses?: Nurse[] }
}) {
  // --- Data and handlers -----------------------------------------------------
  const formatedRequests = useMemo(
    () =>
      doctors?.map(({ id, govId, firstname, lastname, specialties }) => {
        return {
          href: `/institucion/medico/${id}`,
          title: `${firstname} ${lastname}`,
          description: `C.I: ${govId},${specialties?.map((specialty) => ' ' + specialtiesOptions.find((el) => el?.value === specialty.id)?.label)}.`
        }
      }),
    [doctors, specialtiesOptions]
  )

  const nursesRequests = useMemo(
    () =>
      nurses?.map(({ id, govId, firstname, lastname }) => {
        return {
          href: `/institucion/enfermero/${id}`,
          title: `${firstname} ${lastname}`,
          description: `C.I: ${govId}.`
        }
      }),
    [nurses]
  )
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`mx-auto block h-screen w-screen overflow-hidden px-8 ${isIOS() ? 'pb-24 pt-20' : 'pb-24 pt-10'} ${isAndroid() ?? 'pb-64 pt-8'}`}
      data-testid="institution-doctors"
    >
      <div className={`mb-8 flex flex-row justify-between`}>
        <Logo />
        <Menu>
          <MenuButton
            as={IconButton}
            aria-label="Options"
            size="xl"
            fontSize="30px"
            variant="link"
            icon={<HamburgerIcon />}
            onClick={() => null}
          />
          <MenuList>
            <MenuItem>
              <Link href="/institucion/solicitudes">Solicitudes pendientes</Link>
            </MenuItem>
            <MenuItem>
              <Link href="/perfil/institucion">Perfil de la institución</Link>
            </MenuItem>
          </MenuList>
        </Menu>
      </div>
      {formatedRequests && formatedRequests?.length > 0 && (
        <div className="lg:px-80">
          <RequestsList requests={formatedRequests} label="Médicos con acceso" />
        </div>
      )}
      {nursesRequests && nursesRequests?.length > 0 && (
        <div className="lg:px-80">
          <RequestsList requests={nursesRequests} label="Enfermeros con acceso" />
        </div>
      )}
      {formatedRequests?.length === 0 && nursesRequests?.length === 0 && (
        <div
          className="flex h-3/4 w-full flex-col items-center justify-center"
          data-testid="approved-doctors-empty-state"
        >
          <Icon fontSize="xxx-large" as={FaUserDoctor} />
          <Text textAlign="center" mt={4}>
            No hay personal de salud aún.
          </Text>
        </div>
      )}
    </div>
  )
}
