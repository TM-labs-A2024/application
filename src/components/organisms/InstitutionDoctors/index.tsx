import InstitutionDoctors from './InstitutionDoctors'

export default InstitutionDoctors
