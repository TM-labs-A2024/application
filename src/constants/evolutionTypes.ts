export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
