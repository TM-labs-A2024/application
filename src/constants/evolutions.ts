export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
