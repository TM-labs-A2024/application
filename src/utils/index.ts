import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
