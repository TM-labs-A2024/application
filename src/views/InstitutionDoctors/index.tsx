import InstitutionDoctors from '@components/organisms/InstitutionDoctors'
import { Doctor as DoctorType } from '@src/types'
import React from 'react'

export default function InstitutionDoctorsView({ doctors }: { doctors: DoctorType[] }) {
  return <InstitutionDoctors doctors={doctors} />
}
