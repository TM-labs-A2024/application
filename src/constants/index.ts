import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
