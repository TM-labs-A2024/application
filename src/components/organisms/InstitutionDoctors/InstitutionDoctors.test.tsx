import { doctors } from '@src/constants'
import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import InstitutionDoctors from './InstitutionDoctors'

const approvedDoctors = doctors.filter((doctor) => !doctor.pending)

describe('Organisms > InstitutionDoctors test', () => {
  test('The component shows the list of doctors with access if it has at least one doctor', () => {
    render(<InstitutionDoctors doctors={approvedDoctors} />)

    expect(screen.getByText('Médicos con acceso')).toBeInTheDocument()
  })

  test('The component shows the empty state if the list of doctors is empty', () => {
    render(<InstitutionDoctors doctors={[]} />)

    expect(screen.getByTestId('approved-doctors-empty-state')).toBeInTheDocument()
  })

  test("The component allows to go to the institution's profile", () => {
    render(<InstitutionDoctors doctors={approvedDoctors} />)

    expect(screen.getByText('Perfil de la institución')).toHaveAttribute(
      'href',
      '/perfil/institucion'
    )
  })

  test("The component allows to go to the institution's pending requests page", () => {
    render(<InstitutionDoctors doctors={approvedDoctors} />)

    expect(screen.getByText('Solicitudes pendientes')).toHaveAttribute(
      'href',
      '/institucion/solicitudes'
    )
  })

  test('Matches the snapshot', () => {
    render(<InstitutionDoctors doctors={approvedDoctors} />)

    expect(screen.getByTestId('institution-doctors')).toMatchSnapshot()
  })
})
