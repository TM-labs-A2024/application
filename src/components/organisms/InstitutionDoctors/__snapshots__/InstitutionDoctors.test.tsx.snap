// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > InstitutionDoctors test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 pb-24 pt-10 false"
  data-testid="institution-doctors"
>
  <div
    class="mb-8 flex flex-row justify-between"
  >
    <img
      alt="logo"
      data-nimg="1"
      decoding="async"
      height="80"
      loading="lazy"
      src="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75"
      srcset="/_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=256&q=75 1x, /_next/image?url=%2Fstatic%2Fimages%2Flogo-horizontal.png&w=640&q=75 2x"
      style="color: transparent;"
      width="200"
    />
    <button
      aria-controls="menu-list-:rh:"
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="Options"
      class="chakra-button chakra-menu__menu-button css-wpzvz6"
      id="menu-button-:rh:"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="css-r6z5ec"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-1kfu8nn"
        id="menu-list-:rh:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-18esm8n"
          data-index="0"
          id="menu-list-:rh:-menuitem-:ri:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            href="/institucion/solicitudes"
          >
            Solicitudes pendientes
          </a>
        </button>
        <button
          class="chakra-menu__menuitem css-18esm8n"
          data-index="1"
          id="menu-list-:rh:-menuitem-:rj:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            href="/perfil/institucion"
          >
            Perfil de la institución
          </a>
        </button>
      </div>
    </div>
  </div>
  <div
    class="h-full lg:px-80"
  >
    <div
      class="chakra-stack h-full css-8g8ihq"
      data-testid="requests-list"
    >
      <h2
        class="chakra-heading mb-4 css-9f6g39"
      >
        Médicos con acceso
      </h2>
      <div
        class="overflow-auto"
      >
        <ul
          class="css-155za0w"
          role="list"
        >
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/8000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 8000000, Medicina de la educación física y del deporte, Medicina espacial, Endocrinología y nutrición, Alergología, Aparato digestivo, Cardiología.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/9000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 9000000, Medicina espacial, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/10000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 10000000, Medicina intensiva, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/11000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 11000000, Medicina interna, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/12000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 12000000, Medicina legal y forense, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/13000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 13000000, Medicina preventiva y salud pública, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/14000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 14000000, Medicina del trabajo, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/15000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 15000000, Nefrología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/23000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 23000000, Rehabilitación, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/24000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 24000000, Reumatología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/25000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 25000000, Medicina familiar y comunitaria, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/26000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 26000000, Cirugía cardiovascular, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/27000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 27000000, Cirugía general y del aparato digestivo, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/28000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 28000000, Cirugía oral y maxilofacial, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/29000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 29000000, Cirugía ortopédica y traumatología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/30000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 30000000, Cirugía pediátrica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/31000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 31000000, Cirugía plástica, estética y reparadora, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/32000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 32000000, Cirugía torácica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/33000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 33000000, Neurocirugía, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/34000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 34000000, Angiología y cirugía vascular, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/35000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 35000000, Dermatología médico-quirúrgica y venereología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/36000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 36000000, Obstetricia y ginecología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/37000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 37000000, Oftalmología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/38000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 38000000, Otorrinolaringología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/39000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 39000000, Urología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/40000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 40000000, Análisis clínicos, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/41000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 41000000, Anatomía patológica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/42000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 42000000, Bioquímica clínica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/43000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 43000000, Farmacología clínica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/44000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 44000000, Inmunología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/45000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 45000000, Medicina nuclear, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/46000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 46000000, Microbiología y parasitología, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/47000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 47000000, Neurofisiología clínica, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/48000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 48000000, Radiodiagnóstico, Medicina espacial, Endocrinología y nutrición.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`
